### PR TITLE
[DOXIA-716] Update and unify XMLReader creation and configuration

### PR DESCRIPTION
**Pull request by michael-o**
This closes #187

### CODE REVIEW

File: doxia-core/src/test/java/org/apache/maven/doxia/xsd/AbstractXmlValidator.java [hunk 1]
```diff
@@ -157,8 +161,17 @@ protected void setValidate(boolean validate) {
     private XMLReader getXMLReader() {
         if (xmlReader == null) {
             try {
-                xmlReader = XMLReaderFactory.createXMLReader("org.apache.xerces.parsers.SAXParser");
+                SAXParserFactory parserFactory = SAXParserFactory.newInstance();
```
Comment (kwin) [inline doxia-core/src/test/java/org/apache/maven/doxia/xsd/AbstractXmlValidator.java:164]: Is it possible to shared the same method in AbstractXmlValidator and XmlValidator somehow? It seems the same features need to be enabled here.

**Review by kwin — APPROVED** [review]
LGTM

File: doxia-core/src/test/java/org/apache/maven/doxia/xsd/AbstractXmlValidator.java (hunk 1 above)
Comment (michael-o) [reply doxia-core/src/test/java/org/apache/maven/doxia/xsd/AbstractXmlValidator.java:164]: I bet so. Let me evaluate that later this day.

**Review by slachiewicz — APPROVED** [review]
looks good to me. 
While we are here maybe something for parsing and later for validation can be hardened:
https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html

**Comment by michael-o** [comment]
> looks good to me. While we are here maybe something for parsing and later for validation can be hardened: https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html

This should be separate, please file a JIRA issue.

File: doxia-core/src/test/java/org/apache/maven/doxia/xsd/AbstractXmlValidator.java (hunk 1 above)
Comment (michael-o) [reply doxia-core/src/test/java/org/apache/maven/doxia/xsd/AbstractXmlValidator.java]: Done and ran through doxia-sitetools and m-site-p.

File: doxia-core/src/main/java/org/apache/maven/doxia/util/XmlValidator.java [hunk 2]
```diff
@@ -65,57 +86,42 @@ public class XmlValidator {
...
 
     /**
-     * @param hasDtdAndXsd to flag the <code>ErrorHandler</code>.
      * @return an xmlReader instance.
      * @throws SAXException if any
+     * @throws ParserConfigurationException
      */
-    private XMLReader getXmlReader(boolean hasDtdAndXsd) throws SAXException {
+    public XMLReader getXmlReader() throws SAXException, ParserConfigurationException {
         if (xmlReader == null) {
```
Comment (kwin) [inline doxia-core/src/main/java/org/apache/maven/doxia/util/XmlValidator.java:101]: I am not sure if caching is wise here, as SAXParser/SAXParserFactory is not not thread-safe AFAIK (https://docs.oracle.com/cd/E17802_01/webservices/webservices/docs/1.5/api/javax/xml/parsers/SAXParserFactory.html). At least this should be documented in the javadoc that the returned XMLReader needs to be synchronized somehow to be used among multiple threads.

Comment (michael-o) [reply doxia-core/src/main/java/org/apache/maven/doxia/util/XmlValidator.java:101]: No objections, but that would be a behavioral change and should be a separate issue.

**Pull request merged**